jsonrpc2/client.go

```diff
@@ -239,9 +239,15 @@ func (c Client) Notify(serviceMethod string, args interface{}) error {
 // NewClient returns a new Client to handle requests to the
 // set of services at the other end of the connection.
 func NewClient(conn io.ReadWriteCloser) *Client {
-	codec := newClientCodec(conn)
+	codec := NewClientCodec(conn)
 	client := rpc.NewClientWithCodec(codec)
-	return &Client{client, codec.(*clientCodec)}
+	return &Client{client, codec}
```

Comment:
Looks like whole `NewClient` should be simplified to `return NewClientWithCodec(NewClientCodec(conn))`.